skrub/_datetime_encoder.py

```diff
@@ -105,11 +115,39 @@ class DatetimeEncoder(SingleColumnTransformer):
         Add the total number of seconds since the Unix epoch (00:00:00 UTC on 1
         January 1970).
 
+    add_day_of_year : bool, default=False
+        Add the day of year (ordinal day) as an integer in the range 1 to 365 (or
+        366 in the case of leap years). January 1st will be day 1, December 31st
+        will be day 365 on non-leap years.
+
+    add_periodic : bool, default=False
+        Add periodic features with different granularities. By default, use
+        trigonometric (circular) encoding of features. Spline encoding and
+        custom encoders are also supported.
+
+    year_encoding : str, :class:``~CircularEncoder``, :class:``~SplineEncoder`` or None, default="circular"
+        Select the strategy used to encode days in a year. By default, use a
+        CircularEncoder with period=365. If None, no encoding is performed.
```

Comment:
If I follow things right, this is used only if "add_periodic" is True above. We should mention this.